File: app/src/main/java/com/github/willjgriff/ethereumwallet/ui/send/di/SendComponent.java
package com.github.willjgriff.ethereumwallet.ui.send.di;

import com.github.willjgriff.ethereumwallet.di.AppComponent;
import com.github.willjgriff.ethereumwallet.di.ControllerScope;
import com.github.willjgriff.ethereumwallet.ui.send.SendController;

import dagger.Component;

/**
 * Created by Will on 04/02/2017.
 */

@Component(dependencies = AppComponent.class)
@ControllerScope
public interface SendComponent {

	void inject(SendController sendController);
}

File: app/src/main/kotlin/com/github/willjgriff/ethereumwallet/ui/send/di/SendComponent.kt
package com.github.willjgriff.ethereumwallet.ui.send.di

import com.github.willjgriff.ethereumwallet.di.AppComponent
import com.github.willjgriff.ethereumwallet.di.AppInjector
import com.github.willjgriff.ethereumwallet.di.ControllerScope
import com.github.willjgriff.ethereumwallet.ui.send.SendController
import dagger.Component

/**
 * Created by williamgriffiths on 03/05/2017.
 */
@Component(dependencies = arrayOf(AppComponent::class))
@ControllerScope
interface SendComponent {

    fun inject(sendController: SendController)
}

fun SendController.injectPresenter() {
    DaggerSendComponent.builder()
            .appComponent(AppInjector.appComponent)
            .build()
            .inject(this)
}